### ISSUE TITLE
Cross-compiling warning even when consuming package

### ISSUE THREAD

**Issue by ytimenkov** [conan-io/conan#4602]
Conan 1.12.3

I'm trying to package a compiler (gcc) as a tool to be used as a build requirement.

Since it's a compiler for native code it must have both `os_build` (where it runs on, in my case it can be Linux, Windows or MacOS) and `os` (which is a target os for compilation output, only Linux in my case).
Something like
```python
class GccConan(ConanFile):
    name = "gcc_installer"
    settings = "os", "arch", "os_build"
```
When I build it I get warning:
```
gcc_installer/6.3.1@external/master: ************************************************************
gcc_installer/6.3.1@external/master:   This package defines both 'os' and 'os_build' 
gcc_installer/6.3.1@external/master:   Please use 'os' for libraries and 'os_build'
gcc_installer/6.3.1@external/master:   only for build-requires used for cross-building
gcc_installer/6.3.1@external/master: ************************************************************
```
Which may be fine for general case and I can even put into README a note to ignore it.

However same warning appears even when I try to consume it, basically for every package I build and this is not fine (from usability perspective).

I would suggest moving this warning from initialization of `ConanFile` class to somewhere where we actually build package. Or maybe make it a part of linter / hook.

To help us debug your issue please explain:

- [x] I've read the [CONTRIBUTING guide](https://github.com/conan-io/conan/blob/develop/.github/CONTRIBUTING.md).
- [x] I've specified the Conan version, operating system version and any tool that can be relevant.
- [x] I've explained the steps to reproduce the error or the motivation/use case of the question/suggestion.

**Comment by lasote** [comment]
Yes. I agree it is a bit annoying. I will talk with the team to see what can we do.

**Comment by uilianries** [comment]
Why not use `os_buil` and `arch_build` only? Since the package is an installer, it could be built on the native platform and be distributed as well.

**Comment by ytimenkov** [comment]
@uilianries 
How do I define a compiler which produces binary for Windows or Mac but runs on Linux then?

**Comment by uilianries** [comment]
No my idea just building on native platform. But thinking again, you approach is better for this case, since you can build gcc for Windows, by Linux platform.

**Comment by Minimonium** [comment]
It could be moved to a pre_export, hook, isn't it?

**Comment by memsharded** [comment]
Removed, will be released in 1.21
